multiqc/multiqc.py

```diff
@@ -746,6 +746,30 @@ def __rich_measure__(self, console: rich.console.Console, options: rich.console.
         report.runtimes["mods"][run_module_names[mod_idx]] = time.time() - mod_starttime
     report.runtimes["total_mods"] = time.time() - total_mods_starttime
 
+    # Parse software version from separate YAML file if it exists
+    versions_from_file = software_versions.load_versions_from_yaml(config.version_fn_name)
+    if versions_from_file:
+        name_to_module = {module.anchor: module for module in report.modules_output}
+        for software, versions in versions_from_file.items():
+            # Try to find if the software is listed among the executed modules. Unlisted software are still
+            # reported in the `Software Versions` section.
+            module_name = software_versions.find_matching_module(software, name_to_module)
+            if module_name is None:
+                logger.debug(
+                    "No executed modules matches '{}' listed in '{}'.".format(software, config.version_fn_name)
+                )
+                versions = list(set(map(str, versions)))
+                versions.sort(reverse=True)
+                report.software_versions[software] = versions
+            else:
+                module = name_to_module[module_name]
+                for version in versions:
+                    module.add_software_version(str(version))
```

Comment:
The method needs the sample name as a first argument, so it shouldn't work here?

Reply:
You are absolutely right, this got broken in [df602d4](https://github.com/ewels/MultiQC/pull/1927/commits/df602d46f41b087554fb5f3d35432266a39032fa) but should now be fixed.

Reply:
Excellent, thanks for quickly fixing it! I think it also makes sense to swap the version and sample name arguments and make the second one explicitly optional. I pushed the change now just for the sake of merging everything quickly, I hope you don't mind :)